features/keychain/module/keychain.js

```diff
@@ -42,6 +44,34 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
+  hasLockedPrivateKeys() {
+    return this.#lockedPrivateKeys
+  }
+
+  lockPrivateKeys() {
+    this.#lockedPrivateKeys = true
+  }
+
+  unlockPrivateKeys(seeds) {
+    assert(this.#lockedPrivateKeys, 'already unlocked')
+    assert(
+      seeds?.length === Object.values(this.#masters).length,
+      'must pass in same number of seeds'
+    )
+    for (const seed of seeds) {
+      const seedId = getSeedId(seed)
+      assert(!!this.#masters[seedId], 'must pass in existing seed')
+    }
```

Comment:
Need to check that all passed in seeds were different

Reply:
sorry, was this in response to something or just a clarification?

Reply:
Nope, just a task for myself :) Fixed in 25a4f9603a8b8e13c2387b338bc2437182a0b606